BAT/BCParameter.h

```diff
@@ -79,7 +79,7 @@ class BCParameter : public BCVariable
     /**
      * @return Whether parameter is fixed to a value. */
     virtual bool Fixed() const
-    { return fFixed; }
+    { return fFixed or fLowerLimit == fUpperLimit; }
```

Comment:
This is how it used to be, and it an awful way to structure it. It meant that when you fix a parameter you lose information about its range; so when you unfix a parameter you need to again set its range.

Think of this change more like this: if a user sets a zero-sized range, then BAT should be smart enough not to treat the parameter as free, since it cannot be changed. But the parameter isn't actually fixed. It's just stuck.

What I've implemented recreates the behavior we had and avoids buggy behavior we also apparently had, but which I only just discovered.

Reply:
Ok, we'll leave it at that